packages/grafana-schema/src/schema/dashboard/v2alpha1/types.metadata.gen.ts

```diff
@@ -0,0 +1,30 @@
+// Code generated - EDITING IS FUTILE. DO NOT EDIT.
+
+// metadata contains embedded CommonMetadata and can be extended with custom string fields
+// TODO: use CommonMetadata instead of redefining here; currently needs to be defined here
+// without external reference as using the CommonMetadata reference breaks thema codegen.
+export interface Metadata {
```

Comment:
@radiohead what is this type generated from? In frontend implementation we currently rely on https://github.com/grafana/grafana/blob/haris/dash-schemav2/diffing-sort/public/app/features/apiserver/types.ts#L19, I guess we should not anymore and adjust the code so that API clients have etadata typed through these definitions?

Reply:
I think we need to do some work on unifying SDK metadata and Grafana metadata – they are the same shape but Grafana one has some extra well-known annotations and labels.

The same applies to backend – there's a bunch of stuff in https://github.com/grafana/grafana/blob/poc/dashboard-sdk/pkg/apimachinery/utils/meta.go that the SDK doesn't know about.

CC @IfSentient @ryantxu – we had the plan to move utils/meta to SDK, I guess we should act on it now?